src/test/ReactTestUtils.js

```diff
@@ -410,9 +410,11 @@ NoopInternalComponent.prototype = {
 
 var ShallowComponentWrapper = function(element) {
   // TODO: Consolidate with instantiateReactComponent
-  this._debugID = nextDebugID++;
-  var displayName = element.type.displayName || element.type.name || 'Unknown';
-  ReactInstrumentation.debugTool.onSetDisplayName(this._debugID, displayName);
+  if (__DEV__) {
+    this._debugID = nextDebugID++;
+    var displayName = element.type.displayName || element.type.name || 'Unknown';
+    ReactInstrumentation.debugTool.onSetDisplayName(this._debugID, displayName);
```

Comment:
I looked for it again, and this is the only unguarded `debugTool` call in the codebase. I missed it the last time because I didn’t realize TestUtils could be run in production mode.